content/frontend-web/react-typescript/migrate-class-component-to-functional-component.md

```diff
@@ -0,0 +1,163 @@
+---
+title: Migrate Class Component to Functional Component
+showMetadata: false
+editable: true
+showToc: true
+order: 3
+---
+
+# Local State
+
+เราสามารถแปลงจาก [Class Component](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/class_components/) ด้านล่างนี้
+
+```ts
+import React from "react";
+
+interface MyProps {
+  message: string;
+};
+interface MyState {
+  count: number; 
+};
+class App extends React.Component<MyProps, MyState> {
+  state: MyState = {
+    // optional second annotation for better type inference
+    count: 0,
+  };
+  render() {
+    return (
+      <div>
+        {this.props.message} {this.state.count}
+      </div>
+    );
+  }
+}
+```
+
+ให้เป็น [Functional Component](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components) แบบนี้ได้ 
+
+```ts
+import React, { useState } from "react";
+
+interface PropTypes {
+  message: string;
+}
+
+const App = ({ message }: PropTypes) => {
+  const [count, setCount] = useState(0);
+
+  return (
+    <div>
+      {message} {count}
+    </div>
+  );
+};
+export default App;
+```
+
+ทำไมถึงไม่แนะนำ `React.FC` หรือ `React.FunctionComponent`/`React.VoidFunctionComponent` อ่านรายละเอียดใน [Functional Component](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components) 
+
+# Handle Click Event
+
+เราสามารถแปลงจาก [Class Component](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/class_components) ด้านล่างนี้
+
+```ts
+import React from "react";

```

Comment:
Could you please change double quotes to a single quote everywhere in our code examples?

Reply:
Yes, it can